src/components/element-editor/conditional-rendering/conditional-rendering-child.tsx

```diff
@@ -0,0 +1,161 @@
+    onDelete,
+    options,
+    onChange,
+    value,
+}: IProps) => {
+    // =========================================================================
+    // CONST, STATE, REFS
+    // =========================================================================
+
+    const comparatorOptions = [
+        "Equals",
+        "More than",
+        "Less than",
+        "Not equals",
+    ];
+
+    const { focusedElement } = useBuilder();
+
+    // =============================================================================
+    // EVENT HANLDERS
+    // =============================================================================
+
+    const handleChange = (changeType?: string, newValue?: string) => {
+        const updatedValue = { ...value };
+        if (focusedElement && focusedElement.element) {
+            const internalId = focusedElement.element.internalId;
+
+            switch (changeType) {
+                case "fieldKey":
+                    updatedValue[internalId] = {
+                        ...updatedValue[internalId],
+                        fieldKey: newValue,
+                    };
+                    break;
+                case "comparator":
+                    updatedValue[internalId] = {
+                        ...updatedValue[internalId],
+                        comparator: newValue,
+                    };
+                    break;
+                case "value":
+                    updatedValue[internalId] = {
+                        ...updatedValue[internalId],
+                        value: newValue,
+                    };
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        onChange(updatedValue);
+    };
+
+    // =========================================================================
+    // USE EFFECT
+    // =========================================================================
+
+    useEffect(() => {
+        if (value[focusedElement.element.internalId].comparator) {
+            handleChange(
+                "comparator",
+                value[focusedElement.element.internalId].comparator
+            );
+        } else {
+            handleChange("comparator", comparatorOptions[0]);
+        }
+    }, []);
+
+    // =========================================================================
+    // RENDER FUNCTIONS
+    // =========================================================================
+
+    return (
+        <ChildEntry onDelete={onDelete}>
+            <FieldWrapper>
+                <SelectFieldContainer>
+                    <div>
+                        <SelectFieldWrapper
+                            placeholder="Select"
+                            selectedOption={options.find(
+                                (option) =>
+                                    option?.id ===
+                                    value[focusedElement.element.internalId]
+                                        .fieldKey
```

Comment:
need optional chaining otherwise the app will crash if there are no conditional rendering rules applied for the focused element

```suggestion
                                    value[focusedElement.element.internalId]
                                        ?.fieldKey
```

Reply:
have added this to the new changes that i made aft changing the shape of the conditional rendering child object.